src/benchq/resource_estimation/default_pipelines.py

```diff
@@ -233,74 +232,55 @@ def automatic_resource_estimator(
 ) -> ResourceInfo:
     """Pick the appropriate resource estimator based on the size of the program.
 
-    Currently chooses between GraphResourceEstimator and
-    ExtrapolationResourceEstimator and whether or not to use delayed gate synthesis
+    Currently, chooses between GraphResourceEstimator and
+    ExtrapolationResourceEstimator and whether to use delayed gate synthesis
     in the following order:
         1. GraphResourceEstimator without delayed gate synthesis
-        2. GraphResourceEstimator with delayed gate synthesis
-        3. ExtrapolationResourceEstimator without delayed gate synthesis
+        2. ExtrapolationResourceEstimator without delayed gate synthesis
+        3. GraphResourceEstimator with delayed gate synthesis
         4. ExtrapolationResourceEstimator with delayed gate synthesis
 
 
     Args:
-        program (QuantumProgram): program to estimate resources for
-        error_budget (ErrorBudget): budget for the error the program can tolerate
-        delayed_gate_synthesis (bool, optional): whether or not to decompose.
-            The gates into clifford + T before creating graph. Defaults to False.
-
-    Raises:
-        ValueError: if the problem is too big to estimate resources for
+        algorithm_implementation (AlgorithmImplementation): The algorithm to estimate
+            resources for.
+        hardware_model (BasicArchitectureModel): The hardware model to estimate
+            resources for.
+        decoder_model (Optional[DecoderModel], optional): The decoder model to
+            run the algorithm with. Defaults to None and returns no estimate in
+            that case.
 
     Returns:
-        GraphResourceEstimator: an appropriate resource estimator for the problem
+        ResourceInfo: The resources required to run the algorithm.
     """
-    pipeline: Any = None
     assert isinstance(algorithm_implementation.program, QuantumProgram)
 
-    prev_steps = algorithm_implementation.program.steps
+    graph_size = estimate_full_graph_size(algorithm_implementation, False)
+    reduced_graph_size = estimate_full_graph_size(algorithm_implementation, True)
     algorithm_implementation.program.steps = max(DEFAULT_STEPS_TO_EXTRAPOLATE_FROM)
-    if (
-        estimate_full_graph_size(algorithm_implementation, True)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
-        pipeline = partial(
-            run_fast_extrapolation_estimate,
-            steps_to_extrapolate_from=DEFAULT_STEPS_TO_EXTRAPOLATE_FROM,
-        )
-    elif (
-        estimate_full_graph_size(algorithm_implementation, False)
-        < LARGEST_GRAPH_TOLERANCE
-    ):
+
+    if graph_size < 1e7:
```

Comment:
comment: I'm very surprized that the largest graph size decreased? Not sure what happend there, but it makes me very suspicious....

Reply:
It feels like the previous estimates were taken from different problem instances basically.

For hamiltonians Ive used, 1e7 was very-very stable boundary between "this estimations are taking seconds" versus "it might as well take 1 hour" (which happened for some instances 1e7<estimated_size<1e8